phoenix-core/src/it/java/org/apache/phoenix/end2end/salted/SaltedTableVarLengthRowKeyIT.java

```diff
@@ -87,4 +89,36 @@ public void testSelectValueWithPointKeyQuery() throws Exception {
             conn.close();
         }
     }
+
+    @Test
+    public void testSaltedVarbinaryUpperBoundQuery() throws Exception {
```

Comment:
@dbwong, as I commented in Jira, The description for this bug isn't accurate. This is a bug generally for Salted Table regardless of PK. If the PK is "k INTEGER PRIMARY KEY", we actually have the same issue. Could you please add unit test for "k INTEGER PRIMARY KEY" too?